aexpect/client.py

```diff
@@ -855,7 +855,7 @@ def read_until_output_matches(self, patterns, filter_func=lambda x: x,
             data = self.read_nonblocking(internal_timeout,
                                          end_time - time.time())
             if not data:
-                break
+                continue
```

Comment:
Well when the poller returns `poll_status` there have to be something to be read. If there is anything to be read, the `read_nonblocking` should return it and we should proceed, otherwise we should finish, shouldn't we?

Reply:
Hmm, you are right, when the data is ignored (we do `decode(self.encoding, 'ignore')`) therefore it's possible that we receive something but there is nothing returned. Let me do another set of tests...

Reply:
Yes, tested that with a modified `read_nonblocking` and in case we receive only characters that are ignored by encoding it returns immediately. This fix addresses the `read_until_output_matches` part but even the `read_nonblocking` should improve. Anyway the `read_nonblocking` fix won't address this `read_until_output_matches` issue as we are polling here as well so let me merge this and suggest the `read_nonblocking` fix in a separate PR.

Thank you for noticing this, it must have been really hard to find.